spaghetti/util.py

```diff
@@ -718,9 +719,11 @@ def _points_as_gdf(
             pts_dict = pp.snapped_coordinates
 
     # instantiate geopandas.GeoDataFrame
-    pts_list = list(pts_dict.items())
-    points = geopandas.GeoDataFrame(pts_list, columns=[id_col, geom_col])
-    points.geometry = points[geom_col].map(lambda p: Point(p))
+    points = pandas.DataFrame(list(pts_dict.items()), columns=[id_col, geom_col])
+    points["x"] = points[geom_col].map(lambda coord: coord[0])
+    points["y"] = points[geom_col].map(lambda coord: coord[1])
+    points = points.set_geometry(geopandas.points_from_xy(x=points["x"], y=points["y"]))
+    points = points.drop(columns=["x", "y"])
```

Comment:
More along lines of making a 2D numpy array out of `pts_dict.items()` and passing it to `shapely.points`. I can write that if you can provide a small example of `pts_dict` so I don't have figure out how to get to this point :).

Reply:
I wanted to take a stab at it first. How does the new implementation look?

Reply:
FWIW, the `pts_dict` object looks like:

```python
{"a": (1,1), "b": (2,2)}
```